File: packages/live-update/android/src/main/java/io/capawesome/capacitorjs/plugins/liveupdate/classes/events/DownloadBundleProgressEvent.java
package io.capawesome.capacitorjs.plugins.liveupdate.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.liveupdate.interfaces.Result;

public class DownloadBundleProgressEvent implements Result {

    @NonNull
    private final String bundleId;

    private final long downloadedBytes;
    private final long totalBytes;
    private final double progress;

    public DownloadBundleProgressEvent(@NonNull String bundleId, long downloadedBytes, long totalBytes) {
        this.bundleId = bundleId;
        this.downloadedBytes = downloadedBytes;
        this.totalBytes = totalBytes;
        this.progress = (double) downloadedBytes / (double) totalBytes;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("bundleId", bundleId);
        result.put("downloadedBytes", downloadedBytes);
        result.put("progress", progress);
        result.put("totalBytes", totalBytes);
        return result;
    }
}

File: packages/live-update/ios/Plugin/Classes/Events/DownloadBundleProgressEvent.swift
import Foundation
import Capacitor

@objc public class DownloadBundleProgressEvent: NSObject, Result {
    let bundleId: String
    let downloadedBytes: Int64
    let progress: Double
    let totalBytes: Int64

    init(bundleId: String, downloadedBytes: Int64, totalBytes: Int64) {
        self.bundleId = bundleId
        self.downloadedBytes = downloadedBytes
        self.progress = Double(downloadedBytes) / Double(totalBytes)
        self.totalBytes = totalBytes
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["bundleId"] = bundleId
        result["downloadedBytes"] = Int(truncatingIfNeeded: downloadedBytes)
        result["progress"] = progress
        result["totalBytes"] = Int(truncatingIfNeeded: totalBytes)
        return result as AnyObject
    }
}
